lib/importer.py

```diff
@@ -883,46 +947,82 @@ def execImport(handle: int, options: dict):
+                            libtype=plexLibType,
+                            container_start=sectionProgress,
+                            container_size=maxResults,
+                            maxresults=maxResults,
+                            **updatedFilter
+                        )
+                        log(f"discovered {len(updatedPlexItems)} updated items from {mediaProvider2str(mediaProvider)}")
+                        watchedPlexItems = section.search(
+                            libtype=plexLibType,
+                            container_start=sectionProgress,
+                            container_size=maxResults,
+                            maxresults=maxResults,
+                            **watchedFilter
+                        )
+                        log(f"discovered {len(watchedPlexItems)} new watched items from {mediaProvider2str(mediaProvider)}")
 
-                items.append(item)
+                        plexItems = updatedPlexItems
+                        plexItems.extend(
+                            [item for item in watchedPlexItems if item.key not in [item.key for item in plexItems]]
+                        )
 
-            except plexapi.exceptions.BadRequest as e:
-                # Api.convertDateTimeToDbDateTime may return (404) not_found for orphaned items in the library
-                log(
-                    (
-                        f"failed to retrieve item {plexItem.title} with key {plexItem.key} "
-                        f"from {mediaProvider2str(mediaProvider)}: {e}"
-                    ),
-                    xbmc.LOGWARNING)
-                continue
+                    else:
+                        plexItems = section.search(
+                            libtype=plexLibType,
+                            container_start=sectionProgress,
+                            container_size=maxResults,
+                            maxresults=maxResults,
+                        )
+                except plexapi.exceptions.BadRequest as e:
+                    log(f"failed to fetch {mediaType} items from {mediaProvider2str(mediaProvider)}: {e}", xbmc.LOGINFO)
+                    return
+
+                # Update sectionProgressTotal now that search has run and totalSize has been updated
+                try:
+                    sectionProgressTotal = section.totalSize
+
+                    plexItemsProgressTotal = len(plexItems)
+                    for plexItemsProgress, plexItem in enumerate(plexItems):
+                        if xbmcmediaimport.shouldCancel(handle, plexItemsProgress, plexItemsProgressTotal):
+                            return
+
+                        sectionProgress += 1
+
+                        item = Api.toFileItem(plexServer, plexItem, mediaType, plexLibType)
+                        if not item:
+                            continue
+
+                        itemsToImport.append(item)
+                except plexapi.exceptions.BadRequest as e:
```

Comment:
I would minimize the code envelopped by this `try-except` to the following part:
```python
item = Api.toFileItem(plexServer, plexItem, mediaType, plexLibType)
if not item:
    continue

itemsToImport.append(item)
```

Reply:
Gotchya